cognite/client/data_classes/workflows.py

```diff
@@ -49,8 +50,11 @@ class WorkflowUpsert(WorkflowCore):
     Args:
         external_id (str): The external ID provided by the client. Must be unique for the resource type.
         description (str | None): Description of the workflow. Note that when updating a workflow, the description will
-                            always be overwritten also if it is set to None. Meaning if the wokflow already has a description,
+                            always be overwritten also if it is set to None. Meaning if the workflow already has a description,
                             and you want to keep it, you need to provide the description when updating the workflow.
+        data_set_id (int | None): The id of the data set this workflow belongs to.
+                            If a dataSetId is provided, any operations on this workflow, or its versions, executions,
+                            and triggers will require appropriate access to the data set.
```

Comment:
Adding dataset seems quite complex to me. Let's say it is write-protected, will that stop someone with only read access from triggering for example? It seems worth it to change **...will require appropriate access...** to a more verbose explanation! 😄

Reply:
I'm not sure if this belongs here. Data set behavior should be the same for different resources and has its own documentation. I've documented where it's unique: workflows are assigned to data sets, sub-resources or linked ones require the same access.

Reply:
I've added a link to the datasets docs, which does explain the more subtle concepts.